src/main/scala/com/databricks/spark/sql/perf/Benchmarkable.scala

```diff
@@ -93,4 +111,11 @@ trait Benchmarkable {
     val endTime = System.nanoTime()
     (endTime - startTime).toDouble / 1000000
   }
-}
\ No newline at end of file
+
+  protected def measureTime[A](f: => A): (Duration, A) = {
```

Comment:
a `Duration` object to alleviate your concerns about unit. This can be converted to millis, nanos, etc.